includes/admin/js/geolocator.js

```diff
@@ -1,53 +1,194 @@
+		map.addControl( geocoderControl );
+
+		// When the search box regains focus, show the results list again
+		$( '.leaflet-control-mapbox-geocoder-form' ).find( 'input' ).focus( function() {
+			$( this ).parents( '.leaflet-control-mapbox-geocoder' ).find( '.leaflet-control-mapbox-geocoder-results' ).show();
+		} );
+	}
+
+	/*
+	 * Given a result (featureset) from the Mapbox geocoder
+	 * this parses the data and outputs it to our HTML inputs,
+	 * allowing the data to be stored in the database.
+	 *
+	 * It also gives the user a confirmation message.
+	 */
+	function parseLocationData( feature ) {
+		var city;
+		var country;
+		var region;
+
+		// If the context object doesn't exist, we're dealing with a limited dataset. Let's get what we can.
+		if ( ! feature.context ) {
+			if ( feature.id.match( /^country/ ) ) {
+				country = feature.text;
+			} else if ( feature.id.match( /^region/ ) ) {
+				region = feature.text;
+			}
+
+		// Otherwise, we have a full result set, so let's parse its data
+		} else {
+			for ( var i = 0; i < feature.context.length; i++ ) {
+				if ( feature.context[i].id.match( /^place/ ) ) {
+					city = feature.context[i].text;
+				}
+				if ( feature.context[i].id.match( /^country/ ) ) {
+					country = feature.context[i].text;
 				}
-				var lng = response.features[0].center[0];
-				var lat = response.features[0].center[1];
-				$( '#wanderlist-city' ).val( city );
-				$( '#wanderlist-region' ).val( region );
-				$( '#wanderlist-country' ).val( country );
-				$( '#wanderlist-lng' ).val( lng );
-				$( '#wanderlist-lat' ).val( lat );
-				var placeName = response.features[0].place_name;
-
-				// Vatican doesn't return the correct results, so we're going to do it manually
-				if ( 'Città del Vaticano' === city ) {
-					$( '#wanderlist-region' ).val( '' );
-					$( '#wanderlist-country' ).val( 'Vatican City' );
-					placeName = 'Città del Vaticano';
+				if ( feature.context[i].id.match( /^region/ ) ) {
+					region = feature.context[i].text;
 				}
+			}
 
-				$( '#wanderlist-geocoder-message' ).addClass( 'success' );
-				$( '#wanderlist-geocoder-message' ).find( '.place' ).text( placeName );
-			},
-			error: function(response) {
-				$( '#wanderlist-geocoder-message' ).addClass( 'error' );
-				$( '#wanderlist-geocoder-message' ).find( '.place' ).text( 'ERROR' );
+			// If the city hasn't already been set, we assume that we can get it via the feature name
+			if ( ! city ) {
+				city = feature.text;
+			}
+		}
+
+		// Set lat and long
+		var lng = feature.center[0];
+		var lat = feature.center[1];
+
+		// Output our values to our hidden form fields
+		$( '#wanderlist-city' ).val( city );
+		$( '#wanderlist-region' ).val( region );
+		$( '#wanderlist-country' ).val( country );
+		$( '#wanderlist-lng' ).val( lng );
+		$( '#wanderlist-lat' ).val( lat );
+		var placeName = feature.place_name;
+
+		// Since Geocoder results for Vatican aren't accurate, we'll overwrite them
```

Comment:
This feels like it could be its own function (`fixSpecialCaseCities(city)` or something), as I'm sure there will be a few.

Also, adorable hack but :+1: on the idea.